### PR TITLE
DualBootPatcher: More Sony devices

### PR DESCRIPTION
**Pull request by Caio99BR**
Xperia Z1 Support Picked from PR #10

### CODE REVIEW

**Comment by chenxiaolong** [comment]
@Caio99BR Does this actually work? I though some newer Sony devices still had a combined boot/recovery partition. The last time I looked at a Sony ramdisk (though it was for an old Xperia Sola), it had a very weird ramdisk where the boot and recovery ramdisk files were stored in separate `.cpio.gz` files and `/init` decided which one to load.

This caused problems for DualBootPatcher since it couldn't read `fstab.*` and `init.*.rc` and make the appropriate modifications to them.

@chenxiaolong-bot test this please

File: libmbp/devices/sony.cpp [hunk 1]
```diff
@@ -38,6 +38,33 @@ void addSonyDevices(std::vector<Device *> *devices)
...
+    device->setCodenames({ "honami", "C6903", "C6902", "C6906", "C6943" });
+    device->setName("Sony Xperia Z1");
+    device->setBlockDevBaseDirs({ QCOM_BASE_DIR });
+    device->setSystemBlockDevs({ QCOM_SYSTEM, "/dev/block/mmcblk0p23" });
+    device->setCacheBlockDevs({ QCOM_CACHE, "/dev/block/mmcblk0p24" });
+    device->setDataBlockDevs({ QCOM_USERDATA, "/dev/block/mmcblk0p25" });
+    device->setBootBlockDevs({ QCOM_BOOT, "/dev/block/mmcblk0p14" });
+    device->setRecoveryBlockDevs({ QCOM_FOTA_RECOVERY, "/dev/block/mmcblk0p16" });
+    device->setExtraBlockDevs({ QCOM_ABOOT, QCOM_RPM, QCOM_SBL1, QCOM_TZ });
+    devices.push_back(device);
```
Comment (chenxiaolong) [inline libmbp/devices/sony.cpp]: This line caused a build failure: https://jenkins.noobdev.io/job/DualBootPatcher_Build/target=utilities/54/console

The `.` should be `->` :)

**Comment by Caio99BR** [comment]
@chenxiaolong Yeah the Xperia Devices still use Combined Ramdisk, first because Recovery, since Sony deleted it or renamed to FOTAKernel, idk.

https://raw.githubusercontent.com/TeamVee-SS/android_kernel_sony_falconss/cm-11.0/zip-creator/ramdisk/base-ramdisk/sbin/init.sh

Here the init.sh.
I not tested too for two reasons, 2h downloading all again (i have seriously problem of space) second to test your bot, since is more fast.

And @zhuoyang said "tested and working" on PR #10.

**Comment by Caio99BR** [comment]
Ops, this is a bug, the build was triggered.
I think you need to delete comment of "test this"

**Comment by Caio99BR** [comment]
Ah you can change the init.sh script to bypass 

> exec /init

in last line, so DualBootPatcher will edit necessary files and continue with this command.

**Comment by chenxiaolong** [comment]
> Ops, this is a bug, the build was triggered.
> I think you need to delete comment of "test this"

I have no idea why that is happening. I've reported this bug upstream last week, but haven't received a reply yet: https://github.com/jenkinsci/ghprb-plugin/issues/282 Maybe I'll have to fix it myself.

**Comment by chenxiaolong** [comment]
Thanks for the info about `init.sh`. We'll likely have to build a modified ramdisk patcher (https://github.com/chenxiaolong/DualBootPatcher/tree/master/libmbp/src/ramdiskpatchers) that can allow init.sh to run before mbtool. Right now, when a ramdisk is patched:

* `/init` is symlinked to `/mbtool`
* The original `/init` is renamed to `/init.orig`

On boot, when mbtool completes its tasks, it renames `/init.orig` back to `/init` and runs it.

**Comment by chenxiaolong** [comment]
@Caio99BR I've made an initial attempt at adding support for the combined ramdisk here: https://github.com/chenxiaolong/DualBootPatcher/pull/76

Since I've cherry-picked your commits there, I will close this PR.

**Comment by Caio99BR** [comment]
Ok, no problem

**Pull request closed without merging**